ocaml/xapi/message_forwarding.ml

```diff
@@ -2447,35 +2447,31 @@ functor
               forward_migrate_send ()
           )
         in
-        let message_body =
-          match migration_type with
-          | `Live_interpool ->
-              let source_host = Db.VM.get_resident_on ~__context ~self:vm in
-              Printf.sprintf
-                "VM '%s' (uuid: %s) migrated from host '%s' (uuid: %s) to \
-                 another pool"
-                (Db.VM.get_name_label ~__context ~self:vm)
-                (Db.VM.get_uuid ~__context ~self:vm)
-                (Db.Host.get_name_label ~__context ~self:source_host)
-                (Db.Host.get_uuid ~__context ~self:source_host)
-          | `Non_live ->
-              Printf.sprintf "VM '%s' (uuid: %s) migrated non live"
-                (Db.VM.get_name_label ~__context ~self:vm)
-                (Db.VM.get_uuid ~__context ~self:vm)
-          | `Live_intrapool host ->
-              let source_host = Db.VM.get_resident_on ~__context ~self:vm in
-              Printf.sprintf
-                "VM '%s' (uuid: %s) migrated from host '%s' (uuid: %s) to host \
-                 '%s' (uuid: %s)"
-                (Db.VM.get_name_label ~__context ~self:vm)
-                (Db.VM.get_uuid ~__context ~self:vm)
-                (Db.Host.get_name_label ~__context ~self:source_host)
-                (Db.Host.get_uuid ~__context ~self:source_host)
-                (Db.Host.get_name_label ~__context ~self:host)
-                (Db.Host.get_uuid ~__context ~self:host)
-        in
-        create_vm_message ~__context ~vm ~message_body
-          ~message:Api_messages.vm_migrated ;
+        ( if Db.is_valid_ref __context vm then
```

Comment:
If you have this here, then you'd never print anything for cross pool. We could also collect the info from the DB _before_ the migration rather than having all those DB-gets down here.

Reply:
Hmm... but `create_vm_message` will also fail, so we can't actually send the message.

Reply:
we could do something on the destination instead

Reply:
I'm creating a pull request that removes the call for non live and interpool migration

Reply:
This will fix the issue, i'll add the messages back in `migrate_receive` in a different PR.

Reply:
https://github.com/xapi-project/xen-api/pull/4449
If you prefer to revert all of it, i'll add it back with the `migrate_receive` thing in one PR, what do you prefer?

Reply:
We could go with https://github.com/xapi-project/xen-api/pull/4449 for now (even if it looses the non-live message)

Reply:
separate PR for migrate_receive might be better

Reply:
I know but non live can be interpool as well so I'll put it in the `migrate_receive` PR as well